src/react-native/13-making-http-requests/making-http-requests.md

```diff
@@ -106,17 +114,151 @@ If you’ve implemented the solution correctly, the tests will pass when you run
 
 @diff ../../../exercises/react-native/13-http-requests/01-problem/src/screens/StateList/StateList.tsx ../../../exercises/react-native/13-http-requests/01-solution/src/screens/StateList/StateList.tsx only
 
+</details>
+
+## Objective 2: `fetch` cities in a custom hook
+
+### Setup 2
+
+✏️ Update **src/services/pmo/restaurant/hooks.ts** to be:
+
+@diff ../../../exercises/react-native/13-http-requests/01-solution/src/services/pmo/restaurant/hooks.ts ../../../exercises/react-native/13-http-requests/02-problem/src/services/pmo/restaurant/hooks.ts only
+
+✏️ Create **src/services/pmo/restaurant/interfaces.ts** and update it to be:
+
+@diff ../../../exercises/react-native/13-http-requests/01-solution/src/services/pmo/restaurant/interfaces.ts ../../../exercises/react-native/13-http-requests/02-problem/src/services/pmo/restaurant/interfaces.ts only
+
+✏️ Update **src/screens/CityList/CityList.tsx** to be:
+
+@diff ../../../exercises/react-native/13-http-requests/01-solution/src/screens/CityList/CityList.tsx ../../../exercises/react-native/13-http-requests/02-problem/src/screens/CityList/CityList.tsx only
+
+### Verify 2
+
+✏️ Update **src/services/pmo/restaurant/hooks.test.ts** to be:
+
+@diff ../../../exercises/react-native/13-http-requests/01-solution/src/services/pmo/restaurant/hooks.test.ts ../../../exercises/react-native/13-http-requests/02-problem/src/services/pmo/restaurant/hooks.test.ts only
+
+✏️ Update **src/screens/CityList/CityList.test.tsx** to be:
+
+@diff ../../../exercises/react-native/13-http-requests/01-solution/src/screens/CityList/CityList.test.tsx ../../../exercises/react-native/13-http-requests/02-problem/src/screens/CityList/CityList.test.tsx only
+
+### Exercise 2
+
+TODO
+
+### Solution 2
+
+If you’ve implemented the solution correctly, the tests will pass when you run `npm run test`!
+
+<details>
+<summary>Click to see the solution</summary>
+
+✏️ Update **src/services/pmo/restaurant/hooks.ts** to be:
+
+@diff ../../../exercises/react-native/13-http-requests/02-problem/src/services/pmo/restaurant/hooks.ts ../../../exercises/react-native/13-http-requests/02-solution/src/services/pmo/restaurant/hooks.ts only
```

Comment:
this solution has a fallback of `""` for an argument to the `useCities` hook, but the hook does not conditionally skip api call if it's missing. i am not sure how the api service is set up if this will be an issue, or if there will be a case where `state.short` is null, but just wanted to highlight in case we need to cover it

![Screenshot 2024-05-10 at 11 39 49 AM](https://github.com/bitovi/academy/assets/9858612/7cd29797-e00f-418d-9931-bc60ff662247)

![Screenshot 2024-05-10 at 11 41 05 AM](https://github.com/bitovi/academy/assets/9858612/c0d90a34-5971-4804-b14e-b224f09869fd)

maybe wrap the fetch in a try/catch and set error state?

Reply:
i see in objective #3 it's actually wrapped in try/catch so maybe ok to ignore for this objective?